src/listing.rs

```diff
@@ -246,6 +247,20 @@ pub fn directory_listing(
         return Ok(ServiceResponse::new(req.clone(), res));
     }
 
+    // Serve index file. If it doesn't exist, ignore and list content.
+    if let Some(index_path) = index {
```

Comment:
I think we might want to log an error/warning in this case though. If an index was explicitly set but didn't exist, the user might be confused as to why miniserve is still just providing a file list. A log entry will make the user aware that maybe a path is wrong.

Reply:
I had the impression that this is the default and the expected behaviour, may be because of python's `http.server`. But I think a waning is useful!